### ISSUE TITLE
Improvment/scenarios: sequence of scenario

### ISSUE THREAD

**Issue by AlteredCoder** [crowdsecurity/crowdsec#88]
**Is your feature request related to a problem? Please describe.**
I would like to trigger a scenario when a sequence of event occurs

**Example of what you imagine**
I think a configuration file that looks like this can be a solution:

```yaml
type: sequence
name: vist_uri_1_or_2_and_3
sequence:
 - seq_1:
 	type: leaky 
 	filter: evt.Parsed.uri == "1"
 	capacity: 3    # visit minimum 3 products page
 - seq_2:
 	type: trigger  # visit a specific page
 	filter: evt.Parsed_uri == "2"
 - seq_3:
 	type: trigger
 	filter: evt.Parsed_uri == "3"
condition: (seq_1 or seq_2) and seq_3
group_by: ip_addr
timeframe: 60s     # those 3 events should occurs in 60seconds
```

**Comment by FaricaUnknown** [comment]
One question though: Does the sequence has to be in order ?

**Comment by AlteredCoder** [comment]
I think yes

**Comment by opendba** [comment]
This sounds like a great idea and would allow to catch complex attacks.

**Comment by buixor** [comment]
[Conditional buckets](https://doc.crowdsec.net/docs/next/scenarios/format/#type) have been merged in 1.5.0 and solves this need :smile:
